### ISSUE TITLE
Move group() from Seq to Traversable

### ISSUE THREAD

**Issue by danieldietrich** [vavr-io/vavr#2699]
This is a follow up of #2634, see [this comment](https://github.com/vavr-io/vavr/issues/2634#issuecomment-988144279).

* keep  `group()` as an alias for `slideBy(Function.identity())`
* move `group()` from `Seq`to `Traversable`
* change the return type of `group()` to `Iterator`
* Add `group` to the javadoc of the `Traversable` type

**Comment by sam-chakrabarty** [comment]
I'd love to pick this up if we're looking for a contributor. I'm assuming the goal here is to ship this breaking change in v1.0.0? 
@danieldietrich @pivovarit

**Comment by pivovarit** [comment]
Yes, but I'd not implement it now - the focus is to ship `v0.11.0`, and then we'll start thinking about breaking changes
